### PR TITLE
Check schema of variables in templates and validate input variables

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1462

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java [hunk 1]
```diff
@@ -43,29 +45,54 @@ public class ServiceVariablesJsonSchemaValidator {
...
+        }
+        try {
+            String jsonObjectSchemaString = jsonMapper.writeValueAsString(jsonObjectSchema);
+            Locale.setDefault(Locale.ENGLISH);
+            JsonSchemaFactory factory =
+                    JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012);
+            JsonSchema schema = factory.getSchema(jsonObjectSchemaString);
+            String propertyJson = jsonMapper.writeValueAsString(deployProperty);
+            JsonNode jsonNode = jsonMapper.readTree(propertyJson);
+            Set<ValidationMessage> validate = schema.validate(jsonNode);
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java:83]: @swaroopar This validate method was unable to verify required properties. 
image
When the required property `admin_passwd` is missing, the deploy task will be created not thrown exception. 
image
Then the deploy task failed by required variable not set.
image
I think this type of error should be thrown before generating the new task, So I added additional verification logic.

[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java:83]: @baixinsui this is definitely a bug. Then there is an issue in generating our JSON Schema file. 
The idea was that the json schema what we write automatically validates the required fields. We should not do any other extra validation. Would be great if you can check why this schema validation not working. 

ServiceVariablesJsonSchemaGenerator:77

[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java:83]: Ok. I will check it.

[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaValidator.java:59]: @swaroopar Fixed the error validation with JSON Schema.

**Pull request closed without merging**